plugins/modules/git_config_info.py

```diff
@@ -0,0 +1,187 @@
+
+- name: Read a global config from ~/.gitconfig
+  community.general.git_config_info:
+    name: alias.remotev
+    scope: global
+
+- name: Read a project specific config
+  community.general.git_config_info:
+    name: color.ui
+    scope: local
+    path: /etc
+
+- name: Read all global values
+  community.general.git_config_info:
+    scope: global
+
+- name: Read all system wide values
+  community.general.git_config_info:
+
+- name: Read all values of a specific file
+  community.general.git_config_info:
+    scope: file
+    path: /etc/gitconfig
+'''
+
+RETURN = '''
+---
+config_value:
+  description: >
+    When O(name) is set, a string containing the value of the setting in name. If O(name) is not set, empty.
+    If a config key such as V(push.pushoption) has more then one entry, just the first one is returned here.
+  returned: success if O(name) is set
+  type: str
+  sample: "vim"
+
+config_values:
+  description:
+    - This is a dictionary mapping a git configuration setting to a list of its values.
+    - When O(name) is not set, all configuration settings are returned here.
+    - When O(name) is set, only the setting specified in O(name) is returned here.
+      If that setting is not set, the key will still be present, and its value will be an empty.
+  returned: success
+  type: dict
+  sample:
+    core.editor: ["vim"]
+    color.ui: ["auto"]
+    push.pushoption: ["merge_request.create", "merge_request.draft"]
+    alias.remotev: ["remote -v"]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type="str"),
+            path=dict(type="path"),
+            scope=dict(required=False, type="str", default="system", choices=["global", "system", "local", "file"]),
+        ),
+        required_if=[
+            ("scope", "local", ["path"]),
+            ("scope", "file", ["path"]),
+        ],
+        required_one_of=[],
+        supports_check_mode=True,
+    )
+
+    # We check error message for a pattern, so we need to make sure the messages appear in the form we're expecting.
+    # Set the locale to C to ensure consistent messages.
+    module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C', LC_CTYPE='C')
+
+    name = module.params["name"]
+    path = module.params["path"]
+    scope = module.params["scope"]
+
+    run_cwd = path if scope == "local" else "/"
+    args = build_args(module, name, path, scope)
+
+    (rc, out, err) = module.run_command(args, cwd=run_cwd, expand_user_and_vars=False)
+
+    if rc == 128 and "unable to read config file" in err:
+        # This just means nothing has been set at the given scope
+        pass
+    elif rc >= 2:
+        # If the return code is 1, it just means the option hasn't been set yet, which is fine.
+        module.fail_json(rc=rc, msg=err, cmd=" ".join(args))
+
+    output_lines = [line.strip() for line in out.split("\0") if line.strip()]
```

Comment:
The `strip()` shouldn't really be necessary anymore with `--null`, so how about:
```suggestion
    output_lines = out.split("\0") if out else []
```
The `if out else []` is necessary to not get `['']` if `out` is empty.

Reply:
Thx. This makes a lot of sense. I'm not the python expert as you can see :)